pkg/spdx/document.go

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package spdx
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	spdx_json "github.com/spdx/tools-golang/json"
+
+	"github.com/spdx/tools-golang/spdx/common"
+	"github.com/spdx/tools-golang/spdx/v2_2"
+	"github.com/spdx/tools-golang/spdx/v2_3"
+)
+
+const errOpenDoc = "opening SPDX %s document: %w"
+
+var ErrUnknownFormat = fmt.Errorf("unrecognized document format")
+
+type Document_22 v2_2.Document
+type Document_23 v2_3.Document
+
+type Package struct {
+	PackageLicenseConcluded   string
+	PackageExternalReferences []*PackageExternalReference
+	PackageChecksums          []common.Checksum
+	PackageVersion            string
+}
+
+type File struct {
+	Checksums []common.Checksum
+}
+
+func LoadDocument(path string) (Document, error) {
+	var err1 error
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("opening SPDX document: %w", err)
+	}
+
+	doc23, err1 := spdx_json.Load2_3(f)
+	if err1 == nil && doc23 != nil {
+		return documentFromSPDX(doc23)
+	}
+
+	// First, try to open SPDX 2.2
+	doc22, err := spdx_json.Load2_2(f)
+	if err != nil {
+		err := fmt.Errorf(errOpenDoc, "v2.2", err)
+		if err1 != nil {
+			err = fmt.Errorf("%s + opening 2.3: %w ", err.Error(), err1)
+		}
+		return nil, err
+	}
+
+	if doc22 != nil {
+		return documentFromSPDX(doc22)
+	}
+
+	return nil, ErrUnknownFormat
+}
+
+func getFiles(doc interface{}) []File {
+	files := []File{}
+	switch castDoc := doc.(type) {
+	case *Document_22:
+		for _, of := range castDoc.Files {
+			f := File{}
+			f.Checksums = of.Checksums
+			files = append(files, f)
+		}
+	case *Document_23:
+		for _, of := range castDoc.Files {
+			f := File{}
+			f.Checksums = of.Checksums
```

Comment:
This is just a stylistic question (I bet you write more go than I do). Why not add the Checksums property when you make the `File{}` object like `File{Checksums: of.Checksums}`?